src/cdomains/floatDomain.ml

```diff
@@ -283,6 +295,48 @@ module FloatInterval = struct
 
   let ne = eval_int_binop eval_ne
 
+  let true_nonZero_IInt = IntDomain.IntDomTuple.of_excl_list IInt [(Big_int_Z.big_int_of_int 0)]
+  let false_zero_IInt = IntDomain.IntDomTuple.of_int IInt (Big_int_Z.big_int_of_int 0)
+  let unknown_IInt = IntDomain.IntDomTuple.top_of IInt
+
+  let isfinite op =
+    match op with 
+    | Some v -> true_nonZero_IInt
+    | None -> unknown_IInt
+
+  let isinf op =
+    match op with 
+    | Some v -> false_zero_IInt
+    | None -> unknown_IInt
+
+  let isnan = isinf (**currently we cannot decide if we are NaN or +-inf; both are only in Top*)
+
+  let isnormal op =
+    match op with 
+    | Some (l, h) -> 
+      Printf.printf "called isnormal with l=%g h=%g\n" l h;
+      if l >= Float.min_float || h <= (-. Float.min_float) then
+        true_nonZero_IInt
+      else if l > (-. Float.min_float) && h < Float.min_float then
+        false_zero_IInt
+      else 
+        unknown_IInt
+    | None -> unknown_IInt
+
+  (**it seems strange not to return a explicit 1 for negative numbers, but in c99 signbit is defined as: *)
+  (**<<The signbit macro returns a nonzero value if and only if the sign of its argument value is negative.>> *)
+  let signbit op = 
+    match op with
+    | Some (l, h) ->
```

Comment:
you could pull the inside `if` into the match case with different `when` statements